### ISSUE TITLE
can't find raid0 drives

### ISSUE THREAD

**Issue by hunter6756** [MrPig91/PSChiaPlotter#169]
Can't find raid0 drives in both temp and final directories, neither SSDs nor HDDs in raid0 mode.

Also, is that able to add the feature that the plot queue can skip the 'wait for the copy to final directory' phase like Madmax?

**Comment by sours1000** [comment]
Same problem here. Wont find composite ssd raid 0 drives. I have 2 ssd combined in one as dynamic volume using windows harddrive management. I have to specify this disk as a temporary directory by hand with "switch to basic" and everything works, but it does not appear in the drop-down list

**Comment by MrPig91** [comment]
Yeah this plot manager will not see dynamic disks and I still do not plan to add them since they have been deprecated since 2004. Storage pools is what should be used instead, but as sours1000 mentioned you can use the basic plotting feature and supply the folder paths there. 

1. I want to add this but I would have to essentially recreate that functionality in PSChiaPlotter given how it is currently written. 
2. I am unsure how hpool locks the files but I do have the -Force flag already on the command that tries to delete the file.

**Comment by Jacek-ghub** [comment]
@MrPig91 

> "since they have been deprecated since 2004"

I think that this is a wrong statement, or at least can be read as such.

**Comment by MrPig91** [comment]
@Jacek-ghub you are right, I both read they were deprecated in year 2004 instead of windows 2004. Sorry about that.

**Comment by Jacek-ghub** [comment]
@MrPig91 

> "The Dynamic Disks feature is **no longer being developed**. This feature **will be fully replaced by Storage Spaces in a future release**."

So, that is the original quote from MS website.  
1. Windows 10 v. 2004 was introduced in Q1/2020, so that is just one year ago.
2. Dynamic Disks were not really deprecated, just _development_ was deprecated.  On the end-user side, there is not even a warning or a link to a "preferred solution" about any pending changes in Disk Management.
3. There is no mention that any related API (e.g., for PowerShell) was either made obsolescent or obsoleted.  So, that should not affect PSCP at all.
4. There is no statement when Storage Spaces will be fully functional / ready to replace them.  What it also implies that once Storage Spaces are mature enough, they will replace the current plumbing used by Disk Management, so it will not be an issue at all - transparent from the API point of view.

For example, on my system (Z490 BIOS with Intel series 10 CPU) Storage Spaces are not working at all.  I cannot add any NVMe to it (cannot add even one NVMe, regardless whether it was formatted or not).  That further supports that statement that the end-user replacement "**will be in the (unknown) future.**"

As in one of my previous posts about RAID0, one can use local mapping of those RAIDs, and then use PSCP as normal (Advanced), at least with the PSCP version I used at that time.  So, you can look at the PSCP API usage as two fold: 1. discover disks, and 2. manage them.  PSCP doesn't have any issues whatsoever with the second phase (as it is happy with those mapped drives), that implies that only discovery phase would need to be enhanced.  Enhancement can be done in at least two ways: 1. use some other API and navigate it, or 2. do a pre-processing on available drives, and local map those in questions (what currently end-users needs to do, or be educated to do).

I guess, the success of PSCP depends on how many people will use it.  Items like "Dark Mode" (from the road map) can potentially make some existing end-users happy, but lack of RAID support may potentially drive people away.  I would really suggest that this issue will be classified as essential, as such prioritized before cosmetics, 

Although, I understand that my opinion is just that (my opinion), and it doesn't carry any weight, if at all.  I did try to use RAID, and with drive mapping it was working for me, but at the end, it didn't provide any advantages over using those drives as separate disks (thanks to PSCP doing load balancing!!!).  So, potentially a lot of people are trying to use RAID0 in hopes that it gives their setup some edge, where it may actually hinder the plotting speed a bit.  (It is relevant to those that use HDs in RAID mode, but I assume they are using some additional controller.)

Best, Jacek

**Comment by MrPig91** [comment]
@Jacek-ghub 
Yeah whenever I first read up on dynamic disks I think got a different impression, but I do agree with your general sentiment now after reading up on it more. I will look into adding dynamic disk after I add the "Mover" option. I also agree that Dark Mode is nice, but not something I haven't dedicated any time to since it doesn't effect functionality which should take priority as you mentioned. Your detail write-ups have been very informative and helpful when developing PSChiaPlotter further.
